File: _2391_minimum_amount_of_time_to_collect_garbage/main.py
class Solution:
    def garbageCollection(self, garbage: List[str], travel: List[int]) -> int:
        metal = 0
        paper = 0
        glass = 0

        for i in range(len(garbage) - 1, -1, -1):
            for c in garbage[i]:
                if c == "M":
                    metal += 1
                elif c == "P":
                    paper += 1
                else:
                    glass += 1

            if i <= 0:
                continue
            
            if metal != 0:
                metal += travel[i - 1]
            
            if paper != 0:
                paper += travel[i - 1]

            if glass != 0:
                glass += travel[i - 1]

        return metal + paper + glass

File: _2391_minimum_amount_of_time_to_collect_garbage/main.cpp
class Solution {
public:
    int garbageCollection(vector<string>& garbage, vector<int>& travel) {
        int metal = 0;
        int paper = 0;
        int glass = 0;

        for (int i = garbage.size() - 1; i >= 0; i--)
        {
            for (char& c : garbage[i])
            {
                switch (c)
                {
                    case 'M':
                        metal += 1;
                        break;
                    case 'P':
                        paper += 1;
                        break;
                    case 'G':
                        glass += 1;
                        break;
                }
            }

            if (i <= 0)
            {
                continue;
            }

            if (metal != 0)
            {
                metal += travel[i - 1];
            }

            if (paper != 0)
            {
                paper += travel[i - 1];
            }

            if (glass != 0)
            {
                glass += travel[i - 1];
            }
        }

        return metal + paper + glass;
    }
};